src/WebJobs.Script.WebHost/Diagnostics/DeferredLogDispatcher.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class DeferredLogDispatcher : IDeferredLogDispatcher, IDisposable
```

Comment:
I wonder if this can be an `ILoggerProvider` itself registered at the WebHost level? Then we don't need to update the other web-host loggers to forward to this.

When deferring is no longer needed, that provider could return a singleton `ILogger` which `IsEnabled` always returns `false`.

When enabled, the `ILogger` implementations returned from this would call back into their parent (this instance) to buffer the log.

Reply:
Added logger provider